qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,343 @@
+    derivative component. The construction is designed to reduce the frequency spectrum
+    a normal gaussian curve near the |1>-|2> transition, reducing the chance of leakage
+    to the |2> state.
+
+        f(x) = Gaussian + 1j * beta * d/dx [Gaussian]
+             = Gaussian + 1j * beta * (-(x - duration/2) / sigma^2) [Gaussian]
+
+    where 'Gaussian' is:
+        Gaussian(x, amp, sigma) = amp * exp( -(1/2) * (x - duration/2)^2 / sigma^2) )
+
+    Ref:
+        [1] Gambetta, J. M., Motzoi, F., Merkel, S. T. & Wilhelm, F. K.
+        Analytic control methods for high-fidelity unitary operations
+        in a weakly nonlinear oscillator. Phys. Rev. A 83, 012308 (2011).
+        [2] F. Motzoi, J. M. Gambetta, P. Rebentrost, and F. K. Wilhelm
+        Phys. Rev. Lett. 103, 110501 – Published 8 September 2009
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 beta: float):
+        """Initialize the drag command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Drag envelope.
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+            beta: The correction amplitude.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        self._beta = beta
+        super().__init__(duration=duration)
+        self.validate_params()
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    @property
+    def beta(self):
+        return self._beta
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return drag(duration=self.duration, amp=self.amp, sigma=self.sigma,
+                    beta=self.beta, zero_ends=False)
+
+    def validate_params(self) -> None:
+        if abs(self.amp) > 1.:
+            raise PulseError("The amplitude norm must be <= 1, "
+                             "found: {}".format(abs(self.amp)))
+        if self.sigma <= 0:
+            raise PulseError("Sigma must be greater than 0.")
+        if isinstance(self.beta, complex):
+            raise PulseError("Beta must be real.")
+
+        # Find the first maxima
+        #    This eq is derived from taking the derivative of the norm of the drag function
+        #    and solving for the roots. One solution, of course, is x = duration/2.
+        #    The bound on that solution is handled already by self.amp <= 1. The other two
+        #    solutions mirror each other around the center of the pulse and have the same norm,
+        #    so taking the first x value is sufficient.
+        try:
+            argmax_x = (self.duration / 2
+                        - (1 / self.beta) * math.sqrt(self.beta ** 2 * self.sigma ** 2
+                                                      - self.sigma ** 4))
```

Comment:
I think we only need this for the case `beta > sigma`.
Also why not write it 
```
            argmax_x = (self.duration / 2
                        - (1 / self.beta) * self.sigma * math.sqrt(self.beta ** 2 - self.sigma ** 2)
```

Reply:
That's true, when `beta < sigma` we will get a `ValueError`. Explicitly checking `beta > sigma` seemed opaque. I expect half a year from now I might understand it better if there was one procedure that caught the error with an explanation. I'm open to suggestions so we can make it more clear.

I'd thought about it and it didn't seem to save any space on paper. Looking again, it definitely looks nicer the second way, and makes line 293 more obvious. I'll update the equation, thank you:)

Reply:
My reasoning for checking `beta > sigma` is that it reflects the mathematical logic: i.e. the extrema we are concerned about only exists for `beta > sigma` for `beta < sigma` the maximum seems always to be at `duration/2`.

Reply:
I'll switch to this format 👍